datafusion/expr/src/expr_schema.rs

```diff
@@ -251,9 +251,11 @@ impl ExprSchemable for Expr {
                 ref right,
                 ..
             }) => Ok(left.nullable(input_schema)? || right.nullable(input_schema)?),
-            Expr::Like(Like { expr, .. }) => expr.nullable(input_schema),
-            Expr::ILike(Like { expr, .. }) => expr.nullable(input_schema),
-            Expr::SimilarTo(Like { expr, .. }) => expr.nullable(input_schema),
+            Expr::Like(Like { expr, pattern, .. })
```

Comment:
I also verified that there are no other fields (like `pattern` that can be nullable) 👍 

https://docs.rs/datafusion/latest/datafusion/logical_expr/struct.Like.html